### ISSUE TITLE
Batch indices position - discrepancy between documentation and effects of collation functions

### ISSUE THREAD

**Issue by jac99** [StanfordVL/MinkowskiEngine#66]
SparseTensor documentation says:
_From the version 0.3, we will put the batch indices on the **first column** to be consistent with the standard neural network packages_

However, when I used `SparseCollation` function to combine coordinates, features and labels into tensors, it returns a coords tensor with batch indices in the **last column**.

This part of the code in SparseCollation function indeed adds batch indices as the last column:
```
 coords_batch.append(
                torch.cat((torch.from_numpy(coords[batch_id]).int(),
                           torch.ones(num_points, 1).int() * batch_id), 1))
```

The same with `batched_coordinates` function. Looks like this code adds batch indices as the **last column**:
```
bcoords[s:s + cn, :D] = cs
bcoords[s:s + cn, D] = b
```

What is the correct position of batch indices in the sparse tensor? Should it be in the first or in the last column of coordindates matrix?

**Comment by chrischoy** [comment]
Thanks for point this out. We decided to use the batch index at the last column until v0.4 as this is a breaking change for some.

For future reference, please use `ME.utils.bached_coordinates`, `ME.utils.sparse_collate` , or  `ME.utils.SparseCollation`, to make the code compatible with the latest version of the MinkowskiEngine.

**Comment by chrischoy** [comment]
Minkowski Engine has been updated to v0.4 and batch indices are prepended before coordinates. 
However, it is recommended to use `ME.utils.sparse_collate` or `ME.utils.batched_coordinates` to generate batched coordinates. To extract decomposed coordinates, use `coordinates_at` or `decomposed_coordinates_and_features` on a sparse tensor.
